### DISCUSSION TITLE
(baked?) Shadow on the model wont update

### DISCUSSION THREAD

**Discussion by danilokrulj** [google/model-viewer#3595] [Q&A] [answered]
Hello, I am trying to make some type of configurator where you can add/remove model parts. The problem I am facing is that when I remove parts of the model the parts that were not visible appear to have baked shadows like the removed part is still there. The pictures will probably explain it better than me :) 

![image_2022_07_20T15_03_01_898Z](https://user-images.githubusercontent.com/92090684/180017755-19709e2c-dfe0-4ada-bcc8-d4371e9bcfc8.png)

![image_2022_07_20T15_03_22_667Z](https://user-images.githubusercontent.com/92090684/180017858-f074de60-ae14-48e1-aa30-a83cad176aec.png)

**Comment by elalish** [comment, marked as answer]
Ah yes, this is not any shadows created by model-viewer, but the shadowing of your ambient occlusion texture (exported in your GLB). You'll need to update your texture to remove those: https://modelviewer.dev/examples/scenegraph/#createTexturesExample

**Reply by danilokrulj** [reply]
Thank you, I checked the model in blender and it did have an occlusion map, removed it and the problem was solved.

**Reply by danilokrulj** [reply]
Oh and one more thing. When I swap the model parts it sometime doesn't update the viewer until I manually move it with my mouse. Any clues why? I call updateFraming() and updateShadow() on every change on the model, am I missing something?

**Reply by elalish** [reply]
That sounds like a bug. Can you file an issue with a simple repro link?

**Reply by danilokrulj** [reply]
Sure, I am not sure how to do that though :) do I just upload a link to the webpage where I uploaded the project?

**Reply by elalish** [reply]
Sure, that's fine - just make sure to include precise steps that will always make the problem appear, so we can reproduce it. A screen recording would help too.
